carbon-location.html

```diff
@@ -148,8 +148,8 @@
         }
       },
 
-      __computeRoutePath: function(path, hash, useHashAsPath) {
-        return useHashAsPath ? hash : path;
+      __computeRoutePath: function() {
+        return this.useHashAsPath ? this.__hash : this.__path;
```

Comment:
Previously the behavior of this code was very much reliant on the order of effects, and even small changes in things like the order of elements in a template would result in different outcome for some tests.

After this change the results appear to be consistent and as expected.